controllers/awsadapterconfig_controller.go

```diff
@@ -108,7 +108,7 @@ func (r *AWSAdapterConfigReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	callerIdentity, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
 	if err != nil || callerIdentity == nil || callerIdentity.Account == nil {
-		if callerIdentity == nil || callerIdentity.Account == nil {
+		if err == nil {
 			err = fmt.Errorf("callerIdentity nil")
```

Comment:
There should be two if statements. The existing one, and the new one printing error. I think there is a scenario when error is nil, but the callerIdentity and/or callerIdentity.Account is also nil. See the code change in: https://nirmata.slack.com/archives/D04PMMBUS0Z/p1676985624368149

Reply:
Apologies, I missed that case. Thanks for pointing it out!